### PR TITLE
Fort Insanity fixes and improvements

### PR DESCRIPTION
**Pull request by Halonexus**
## About The Pull Request
Fixes the  fort insanity ai. Makes the fort insanity ai more robust in terms of pathing, damaging and destroying objects in their path, and being able to recover weapons after being disarmed.
Fort insanes can now also fire automatic guns at a decent rate and can multihit with fast melee weapons.
They can also target mechas now.
Additionally makes unarmed human attacks deal red damage instead of melee damage, and allows unarmed attack hit chance to scale with maxhealth, fist damage scales with fort and justice.
Added ego weapon stat check and attack cooldowns when attacking mechas and other objects.
Small runtime fixes for goldrush weapon and pinocchio weapon.

## Why It's Good For The Game
Fort insanes actually do stuff now instead of standing around swapping hands. Standardized the unarmed damage to work with our armor and large healthpools. 

## Changelog
:cl:
fix: fort insane ai works
tweak: ego weapons now have stat checks and attack cooldowns when attacking objects
/:cl:

### CODE REVIEW

File: code/datums/ai/sanity/_sanityloss_controller.dm [hunk 1]
```diff
@@ -155,48 +177,138 @@
...
 			pathing_attempts = 0
-
+			return
 	return TRUE
 
+/turf/proc/reachableTurftestWithMobs(caller, turf/T, ID, simulated_only)
+	if(T && !T.density && !(simulated_only && SSpathfinder.space_type_cache[T.type]) && !LinkBlockedWithAccess(T,caller, ID))
+		for(var/mob/living/L in T)
+			if(L.density)
+				return FALSE
```
Comment (LanceSmites328) [inline code/datums/ai/sanity/_sanityloss_controller.dm:192]: More appropriate would be checking CanPass proc here from what I recall.

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: code/datums/ai/sanity/_sanityloss_controller.dm (hunk 1 above)
Comment (Halonexus) [reply code/datums/ai/sanity/_sanityloss_controller.dm]: yea thats much better

File: code/datums/ai/sanity/_sanityloss_controller.dm [hunk 2]
```diff
@@ -188,12 +188,7 @@
 /turf/proc/reachableTurftestWithMobs(caller, turf/T, ID, simulated_only)
 	if(T && !T.density && !(simulated_only && SSpathfinder.space_type_cache[T.type]) && !LinkBlockedWithAccess(T,caller, ID))
 		for(var/mob/living/L in T)
-			if(L.density)
-				return FALSE
-		for(var/obj/machinery/M in T)
-			if(istype(M, /obj/machinery/door))
-				continue
-			if(M.density)
+			if(!L.CanPass(caller, T))
```
Comment (LanceSmites328) [inline code/datums/ai/sanity/_sanityloss_controller.dm:191]: This no longer counts for machinery in the path, such as doors, like this. The same proc would work for doors as and objects in general here as well. Actually it'd work for all atoms in the turf if you wanted to check that.

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: code/datums/ai/sanity/_sanityloss_controller.dm (hunk 2 above)
Comment (Halonexus) [reply code/datums/ai/sanity/_sanityloss_controller.dm:191]: LinkBlockedWithAccess already does all the object checks

**Review by LanceSmites328 — DISMISSED** [review]

**Review by Kitsunemitsu — APPROVED** [review]
Has been tested on the server for months.
Excellent PR.

**Pull request merged**